app/models/miq_task.rb

```diff
@@ -295,6 +296,17 @@ def process_cancel
     end
   end
 
+  def attributes_hash_when_state_change(state, attributes = {})
```

Comment:
Should this be a private method?  You can still run specs on it, just mark the specs as `describe "#attributes_hash_when_state_change (private)" do`